protocol/genesis/ssv/validator/msgqueue_consumer.go

```diff
@@ -142,6 +142,18 @@ func (v *Validator) ConsumeQueue(logger *zap.Logger, msgID genesisspectypes.Mess
 			logger.Error("❗ got nil message from queue, but context is not done!")
 			break
 		}
+
+		msgLogger := logger
+
+		slot, err := msg.Slot()
+		if err == nil {
+			msgLogger = msgLogger.With(fields.DutyID(fields.FormatDutyID(v.BeaconNetwork.EstimatedEpochAtSlot(slot), slot, msgID.GetRoleType().String(), v.Index)))
+		} else {
+			v.logMsg(msgLogger, msg, "⚠️ could read slot from message",
+				fields.MessageType(spectypes.MsgType(msg.SSVMessage.MsgType)),
+				zap.Error(err))
+		}
+
 		lens = append(lens, q.Q.Len())
 		if len(lens) >= 10 {
 			logger.Debug("📬 [TEMPORARY] queue statistics",
```

Comment:
should we use msgLogger here as well?

Reply:
no because its just statistics about the queue and not related to the msg.

Reply:
Maybe we can use `msgLogger` right before calling the handler to make it less confusing

Reply:
It might be better to move this modification into the inner handler function and build the dutyID inside. For example:
```go
slot, err := msg.Slot()
if err != nil {
    return fmt.Errorf("could not get slot from message: %w", err)
}
dutyID := fields.FormatDutyID(v.BeaconNetwork.EstimatedEpochAtSlot(slot), slot, msg.MsgID.GetRoleType().String(), v.Index)
logger = logger.With(fields.DutyID(dutyID))
```